packages/react/src/ReactCurrentActingRendererSigil.js

```diff
@@ -8,12 +8,12 @@
  */
 
 /**
- * Used by act() to track whether you're outside an act() scope.
- * We use a renderer's flushPassiveEffects as the sigil value
- * so we can track identity of the renderer.
+ * We maintain a 'stack' of renderer specific sigils
+ * corresponding to act() calls, so we can track whenever an update
+ * happens inside/outside of one.
  */
 
 const ReactCurrentActingRendererSigil = {
-  current: (null: null | (() => boolean)),
+  current: ([]: Array<{}>),
```

Comment:
This is conceptually a Set, not an Array, so (assuming this is the approach we go with) let's make it an Array.

Reply:
I started this with a Set, until I realized it's *not* one. It's a stack. Consider when leaving an act() scope, how could I know if the sigil has to be removed from the set?